### DISCUSSION TITLE
Timestamp with system datetime does not match Windows datetime

### DISCUSSION THREAD

**Discussion by lachioma** [bonsai-rx/bonsai#1642] [Q&A]
Hello

I use this sequence at the beginning of my workflow to get a date time string. I include this string in the filename of all the files my workflow generates.

![image](https://github.com/bonsai-rx/bonsai/assets/29898879/a5427d66-9b0c-4b45-96f7-6ef567829ec2)

However, I noticed that if I keep Bonsai instance open for long time (days), and I then start the workflow, this timestamp is off by minutes compared to the Windows system date time. 
Is this behavior expected? How can I fix it (= get a timestamp that always matches Windows date time)?

Many thanks for any help!

**Comment by bruno-f-cruz** [comment]
Never seen this behavior before. Interesting. Naively I would suggest trying DateTime.Now in an ExpressionTransform or csharp script.

**Reply by lachioma** [reply]
I changed the sequence and using now an `ExpressionTransform` (see screenshot). I will post an update whether it works or not.

![image](https://github.com/bonsai-rx/bonsai/assets/29898879/e59178f9-50ea-435b-a36f-baca11c768f5)

If anyone wonders, the `Format` node contains the following: `{0:yyyyMMdd_HHmmss}`. It just formats the timestamp as "yyyyMMdd_HHmmss".
